test/modules/aws-cloudfront-integration.ts

```diff
@@ -10,187 +12,270 @@ import {
   execComposeDown,
   runSync,
   getPrefix,
-} from "../helpers";
+} from '../helpers';
 
 const prefix = getPrefix();
-const dbAlias = "cloudfront";
+const dbAlias = 'cloudfront';
 const callerReference = `${prefix}-caller`;
+const s3CallerReference = `s3-${prefix}-caller`;
 const originId = `${prefix}-origin-id`;
-const behavior = { TargetOriginId: originId, ViewerProtocolPolicy: 'allow-all', CachePolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6" };
+const s3OriginId = `${prefix}-s3-origin-id`;
+const bucket = `${prefix}-bucket`;
+
+const behavior = {
+  TargetOriginId: originId,
+  ViewerProtocolPolicy: 'allow-all',
+  CachePolicyId: '658327ea-f89d-4fab-a63d-7e88639e58f6',
+};
 const behaviorString = JSON.stringify(behavior);
-const origins = [{ DomainName : 'www.google.com', Id: originId, CustomOriginConfig: {HTTPPort: 80, HTTPSPort: 443, "OriginProtocolPolicy": "https-only"}}]
+const origins = [
+  {
+    DomainName: 'www.google.com',
+    Id: originId,
+    CustomOriginConfig: { HTTPPort: 80, HTTPSPort: 443, OriginProtocolPolicy: 'https-only' },
+  },
+];
 const originsString = JSON.stringify(origins);
+const s3behavior = {
+  TargetOriginId: s3OriginId,
+  ViewerProtocolPolicy: 'allow-all',
+  CachePolicyId: '658327ea-f89d-4fab-a63d-7e88639e58f6',
+};
+const s3behaviorString = JSON.stringify(s3behavior);
 
 const apply = runApply.bind(null, dbAlias);
 const sync = runSync.bind(null, dbAlias);
 const query = runQuery.bind(null, dbAlias);
 const install = runInstall.bind(null, dbAlias);
 const uninstall = runUninstall.bind(null, dbAlias);
-const modules = ["aws_cloudfront"];
+const modules = ['aws_cloudfront'];
+
+const region = process.env.AWS_REGION ?? '';
+const accessKeyId = process.env.AWS_ACCESS_KEY_ID ?? '';
+const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY ?? '';
+const s3Client = new S3({ credentials: { accessKeyId, secretAccessKey }, region });
 
 jest.setTimeout(620000);
-beforeAll(async () => await execComposeUp());
-afterAll(async () => await execComposeDown());
 
-describe("Cloudfront Integration Testing", () => {
-  it("creates a new test db", (done) =>
-    void iasql
-      .connect(dbAlias, "not-needed", "not-needed")
-      .then(...finish(done)));
+const createS3Bucket = async () => {
+  const result = await s3Client.createBucket({
```

Comment:
unfortunately s3 module is not exposing the Location field, it will need a modification in s3 module, then revisit it. Will be creating the issue for it

Reply:
This should probably be updated with this new info here? https://discord.com/channels/920684574231646229/949065192275316736/1017063777226727544

I think using the `aws_s3` module will work.